zeeguu/api/utils/route_wrappers.py

```diff
@@ -19,16 +24,37 @@ def with_session(view):
     def wrapped_view(*args, **kwargs):
         try:
             session_uuid = flask.request.args["session"]
-            session_object = Session.find(session_uuid)
-            if session_object is None:
+            user_id, timeout = SESSION_CACHE.get(
+                session_uuid,
+                (
+                    None,
+                    datetime.now() - timedelta(0, SESSION_CACHE_EXPIRE),
+                ),
+            )
+            if datetime.now() > timeout:
```

Comment:
this is too complicated for me :(

by timeout I understand a duration, like 60s. so maybe this var should be named something like `session_expiry_time`?

Reply:
Indeed. Personally, I just see a timeout as when it should expire, but then `session_expiry_time `fits for both! Changed!